lib/integrations/externalcloudaudit/bootstrap.go

```diff
@@ -0,0 +1,405 @@
+	PutObjectLockConfiguration(ctx context.Context, params *s3.PutObjectLockConfigurationInput, optFns ...func(*s3.Options)) (*s3.PutObjectLockConfigurationOutput, error)
+	// Sets the versioning state of an existing bucket.
+	PutBucketVersioning(ctx context.Context, params *s3.PutBucketVersioningInput, optFns ...func(*s3.Options)) (*s3.PutBucketVersioningOutput, error)
+	// Creates a new lifecycle configuration for the bucket or replaces an existing lifecycle configuration.
+	PutBucketLifecycleConfiguration(ctx context.Context, params *s3.PutBucketLifecycleConfigurationInput, optFns ...func(*s3.Options)) (*s3.PutBucketLifecycleConfigurationOutput, error)
+}
+
+// Creates a bucket with the given name in the given region.
+// Long Term Storage buckets have the following properties:
+// * Bucket versioning enabled
+// * Object locking enabled with Governance mode and default retention of 4 years
+// * Object ownership set to BucketOwnerEnforced
+// * Default SSE-S3 encryption
+func createLTSBucket(ctx context.Context, clt createBucketClient, bucketName string, region string) error {
+	err := createBucket(ctx, clt, bucketName, region, true)
+	if err != nil {
+		if !trace.IsAlreadyExists(err) {
+			return trace.Wrap(err)
+		}
+	}
+
+	_, err = clt.PutObjectLockConfiguration(ctx, &s3.PutObjectLockConfigurationInput{
+		Bucket: &bucketName,
+		ObjectLockConfiguration: &s3types.ObjectLockConfiguration{
+			ObjectLockEnabled: s3types.ObjectLockEnabledEnabled,
+			Rule: &s3types.ObjectLockRule{
+				DefaultRetention: &s3types.DefaultRetention{
+					Years: defaultRetentionYears,
+					// Modification is prohibited without BypassGovernancePermission iam permission
+					Mode: s3types.ObjectLockRetentionModeGovernance,
+				},
+			},
+		},
+	})
+	if err != nil {
+		return trace.Wrap(awsutil.ConvertS3Error(err))
+	}
+
+	return nil
+}
+
+// createTransientBucket is similar to createLTSBucket however object locking is not enabled and instead a lifecycle
+// policy is created that cleans up transient storage:
+// * Query results expire after 1 day
+// * Large payloads expire after 4 days
+// * DeleteMarkers, NonCurrentVersions and IncompleteMultipartUploads are also removed
+func createTransientBucket(ctx context.Context, clt createBucketClient, bucketName string, region string) error {
+	err := createBucket(ctx, clt, bucketName, region, false)
+	if err != nil {
+		if !trace.IsAlreadyExists(err) {
+			return trace.Wrap(err)
+		}
+	}
+
+	_, err = clt.PutBucketLifecycleConfiguration(ctx, &s3.PutBucketLifecycleConfigurationInput{
+		Bucket: &bucketName,
+		LifecycleConfiguration: &s3types.BucketLifecycleConfiguration{
+			Rules: []s3types.LifecycleRule{
+				{
+					Status: s3types.ExpirationStatusEnabled,
+					ID:     aws.String("ExpireLargePayloads"),
+					Expiration: &s3types.LifecycleExpiration{
+						Days: 4,
+					},
+					Filter: &s3types.LifecycleRuleFilterMemberPrefix{
+						Value: "/large_payloads",
+					},
+				},
```

Comment:
I think large events will stay on our cloud, similar to how the SQS queue stays on our cloud, so we don't drop any events in case of a temporary loss of connection to customer infra

Reply:
That's right and it explains the missing large payload bucket URI in external cloud audit spec. Thanks!

Reply:
Removed in [9fe0ad3](https://github.com/gravitational/teleport/pull/33272/commits/9fe0ad3f69c4d987801d917d761216168d0716ec)